plugins/woocommerce/src/Blocks/BlockTypes/FeaturedProduct.php

```diff
@@ -22,7 +22,7 @@ protected function get_item( $attributes ) {
 		$id = absint( $attributes['productId'] ?? 0 );
 
 		$product = wc_get_product( $id );
-		if ( ! $product ) {
+		if ( ! $product || 'publish' !== $product->get_status() ) {
```

Comment:
I didn't think that we need to account for some users being able to view products which might not be published. Ie: the admin of the site can see private or draft products, so it would be nice if the Featured Product block behaves the same.

Adding one extra check via `current_user_can()` should be enough:

```suggestion
		if ( ! $product || ( 'publish' !== $product->get_status() && ! current_user_can( 'read_product', $id ) ) ) {
```